clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy

```diff
@@ -483,7 +483,13 @@ class BasicAmazonDeployHandler implements DeployHandler<BasicAmazonDeployDescrip
         device.size = volumeSize
         device.volumeType = volumeType
         device.snapshotId = snapshotId
-        device.encrypted = encrypted
+        if (snapshotId != null) {
```

Comment:
could this be

```
if (snapshotId == null) {
  // aws only allows specifying the encrypted flag when a snapshotId has not been provided 
  device.encrypted = encrypted
}
```

(this way you'd avoid having to write both the if/else)

Reply:
sure 👍 for cleanup